### DISCUSSION TITLE
Problem using TSchema from an internally library

### DISCUSSION THREAD

**Discussion by cesxhin** [sinclairzx81/typebox#1115] [Q&A] [answered]
The library I made myself is typed.
The idea is to use a basic schema and then implement my custom fields in various projects.

The problem is that the ".d.ts" files do not have enough information.
I have this message: `Property '[Kind]' is missing in type 'TObject<{ [...] }>' but required in type 'TSchema'.ts(2741)`

in the .d.ts file generated like this:
`
export declare const schemaDefault: import("@sinclair/typebox").TObject<{
    exaple: import("@sinclair/typebox").TString;
    exaple_2: import("@sinclair/typebox").TString;
   [....]
}>;
`

How can I solve it?

**Comment by sinclairzx81** [comment]
@cesxhin Hi,

If you are exporting TypeBox types from your library. You will need to ensure the application is using the same version of TypeBox as your library is. If the versions are different, TypeScript will flag this as an error (this is because the Kind is a unique symbol, and symbols are tracked per package version)

If using TypeBox as a dependency, it is usually good to specify it as a `peerDependency`. This way the user (or application) can install the version they want, and your library will get the same version.

For a reference project, take a look at https://github.com/fastify/fastify-type-provider-typebox 

Does this help?
S

**Reply by cesxhin** [reply]
@sinclairzx81
I tried using "peerDependency" but nothing changed.

In my opinion it's not a version problem, there's something else that when I run "tsc". I wrong something configuration of tsconfig?

**Reply by paulius-valiunas** [reply, marked as answer]
@cesxhin the error just means you have two instances of Typebox imported somewhere in your dependency tree. We recently ran into the same issue and we made sure we only have one version of it in our lock file, but the issue was still there. The problem turned out to be that we were importing it as an ES module (`import` keyword) in one place, and as a CommonJs module (`require`) in another. Apparently symbols are tracked separately for CJS and ESM, not only for different versions.
